go/pinentry/pinentry_osx.go

```diff
@@ -78,11 +78,21 @@ func (pi *pinentryInstance) shouldStoreSecret(info pinentrySecretStoreInfo) bool
 	// up saying that the client wants to access the user's
 	// keychain. But this will do for now until we write our own
 	// pinentry.
-	attributes := kc.GenericPasswordAttributes{
-		ServiceName: pinentryServiceName,
-		AccountName: string(info),
+	query := keychain.NewItem()
+	query.SetSecClass(keychain.SecClassGenericPassword)
+	query.SetService(pinentryServiceName)
+	query.SetAccount(string(info))
+	query.SetMatchLimit(keychain.MatchLimitAll)
+	err := keychain.DeleteItem(query)
```

Comment:
You could use this shortcut instead:
err := keychain.DeleteGenericPasswordItem(pinentryServiceName, string(info))